bpf/tests/bpf-prog-test-run/bpf-prog-test-run.go

```diff
@@ -0,0 +1,494 @@
+		}).Fatalf("error creating file: %v", err)
+	}
+	if err = outF.Truncate(0); err != nil {
+		log.WithFields(log.Fields{
+			"filename": fnameOut,
+		}).Fatalf("error truncating file: %v", err)
+	}
+	defer outF.Close()
+
+	if _, err = io.Copy(outF, osF); err != nil {
+		log.WithFields(log.Fields{
+			"filename": fnameOut,
+		}).Fatalf("error copying file: %v", err)
+	}
+
+	zero := [4 * 4]byte{}
+	for symName, symOff := range mapSymbols {
+		idOff := symOff + (5 * 4)
+		_, err = outF.WriteAt(zero[:], int64(idOff))
+		if err != nil {
+			log.WithFields(log.Fields{
+				"filename": fnameOut,
+			}).Fatalf("error patching file: %v", err)
+		}
+		log.WithFields(log.Fields{
+			"map": symName,
+		}).Debug("zeroed last fields")
+	}
+
+	return nil
+
+}
+
+func dumpDebugMessages(eventsReader *perf.Reader) error {
+	fmt.Printf("Log Messages:\n")
+	for {
+		record, err := eventsReader.Read()
+		if err != nil {
+			return fmt.Errorf("error reading event buffer: %v\n", err)
+		}
+
+		dm := monitor.DebugMsg{}
+		reader := bytes.NewReader(record.RawSample)
+		if err := binary.Read(reader, byteorder.Native, &dm); err != nil {
+			return fmt.Errorf("Error while parsing debug message: %v\n", err)
+		}
+
+		// special record values set from our bpf programs to mark their termination
+		if dm.SubType == monitor.DbgUnspec && dm.Arg1 == 0xe3d && dm.Arg2 == 0xe3d {
+			break
+		}
+		dm.Dump("")
+	}
+
+	return nil
+}
+
+func testCt4Rst(spec *ebpf.Collection) error {
+
+	eventsReader, err := perf.NewReader(spec.Maps["test_events_map"], 1024*4096)
+	if err != nil {
+		log.WithError(err).Fatal("Cannot initialise BPF perf ring buffer sockets")
+	}
+	defer func() {
+		eventsReader.Close()
+	}()
+
+	packetSyn := gopacket.NewSerializeBuffer()
+	gopacket.SerializeLayers(packetSyn, gopacket.SerializeOptions{},
+		&layers.Ethernet{
+			SrcMAC:       net.HardwareAddr{1, 0, 0, 3, 0, 10},
+			DstMAC:       net.HardwareAddr{1, 0, 0, 3, 0, 20},
+			EthernetType: layers.EthernetTypeIPv4,
+		},
+		&layers.IPv4{
+			Version:  4,
+			SrcIP:    net.IP{10, 3, 0, 10},
+			DstIP:    net.IP{10, 3, 0, 20},
+			Protocol: layers.IPProtocolTCP,
+			IHL:      5,
+		},
+		&layers.TCP{
+			SrcPort:    3010,
+			DstPort:    3020,
+			DataOffset: 5,
+			SYN:        true,
+		},
+		gopacket.Payload([]byte{'p', 'i', 'z', 'z', 'a', '!', ' ', ':', '-', ')'}),
+	)
+
+	packetRst := gopacket.NewSerializeBuffer()
+	gopacket.SerializeLayers(packetRst, gopacket.SerializeOptions{},
+		&layers.Ethernet{
+			DstMAC:       net.HardwareAddr{1, 0, 0, 3, 0, 10},
+			SrcMAC:       net.HardwareAddr{1, 0, 0, 3, 0, 20},
+			EthernetType: layers.EthernetTypeIPv4,
+		},
+		&layers.IPv4{
+			Version:  4,
+			DstIP:    net.IP{10, 3, 0, 10},
+			SrcIP:    net.IP{10, 3, 0, 20},
+			Protocol: layers.IPProtocolTCP,
+			IHL:      5,
+		},
+		&layers.TCP{
+			DstPort:    3010,
+			SrcPort:    3020,
+			DataOffset: 5,
+			RST:        true,
+		},
+		gopacket.Payload([]byte{'p', 'i', 'z', 'z', 'a', '!', ' ', ':', '-', ')'}),
```

Comment:
Not a big deal, but was wondering why not just make a string "pizza! :-)" and then cast to `[]byte`?

Reply:
fixed